File: back/Dto/ProductDto/ProductInputPostDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace back.Dto.ProductDto
{
    public class ProductInputPostDto
    {
        public long? ProductId { get; set; }
        [Required]
        public string ProductName { get; set; }
        [Required]
        public bool Status { get; set; }
        [Required]
        public DateTime DateIn { get; set; }
        [Required]
        public DateTime DateOut { get; set; }
        [Required]
        public int Quantity { get; set; }
        [Required]
        public long CategoryId { get; set; }
    }
}

File: front/src/app/_Dto/ProductDto/ProductInputPostDto.ts
export class ProductInputPostDto {
    productId: number = 0;
    productName: string;
    status: boolean;
    dateIn: any;
    dateOut: any;
    quantity: any;
    categoryId: any;
    constructor(categoryId: any) {
        this.categoryId = categoryId;
    }
}